### PR TITLE
webapp: support to retrieve creds from acr

### LINKED ISSUE #3597: Azure WebApp config container set should not require creds for ACR repos
When configuring an Azure WebApp to run a container, the registry URI, password, and login/user are required. This pretty cumbersome when working with ACR, since you need to write several commands to fetch the credential values. Instead, the command should accept an image name and (optional) resource ID or name for an ACR repo.

### PR DESCRIPTION
**Pull request by yugangw-msft**
Fix #3597

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -319,10 +319,22 @@ def update_container_settings(resource_group_name, name, docker_registry_server_
     settings = []
     if docker_registry_server_url is not None:
         settings.append('DOCKER_REGISTRY_SERVER_URL=' + docker_registry_server_url)
+
+    if (not docker_registry_server_user and not docker_registry_server_password and
+            docker_registry_server_url and '.azurecr.io' in docker_registry_server_url):
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: The sniffing mechanism on `.azurecr.io` was communicated to ACR team. As ACR has no support in sovereign clouds, we are fine for now. I will open a tracking bug to get it into CLI cloud context when there are ready.

**Comment by yugangw-msft** [comment]
@derekbekoe, could you please take a look at this PR?

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -319,10 +319,22 @@ def update_container_settings(resource_group_name, name, docker_registry_server_
...
+        logger.warning('No credential was provided to access Azure Container Registry. Trying to look up...')
+        parsed = urlparse(docker_registry_server_url)
+        registry_name = (parsed.netloc if parsed.scheme else parsed.path).split('.')[0]
+        try:
+            docker_registry_server_user, docker_registry_server_password = _get_acr_cred(registry_name)
+        except Exception as ex:  # pylint: disable=broad-except
+            logger.warning("Error:'%s'", ex)  # consider to throwing if needed
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: `logger.warning` and `"Error"` in the actual message is confusing.
Can you use `logger.error` instead? Or, if this is a fatal error, raise CLIError.

Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Done, i removed the `Error`

**Pull request merged**